src/tunnel.c

```diff
@@ -370,8 +370,12 @@ static int tcp_connect(struct tunnel *tunnel)
 		char *proxy_host, *proxy_port;
 		// protect the original environment from modifications
 		env_proxy = strdup(env_proxy);
+		if (env_proxy == NULL) {
+			log_error("strdup: %s\n", strerror(errno));
+			goto err_strdup;
+		}
 		// get rid of a trailing slash
-		if (env_proxy[strlen(env_proxy) - 1] == '/')
+		if (*env_proxy && env_proxy[strlen(env_proxy) - 1] == '/')
```

Comment:
but here it is really cosmetic then, but probably requested by coverity

Reply:
Coverity fears _getenv_() might return an empty string "" but I don't know whether this can happen. The documentation doesn't specify that.